src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -391,6 +407,23 @@ private static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<cha
             return CombineNoChecksInternal(first, second, third);            
         }
 
+        private static string CombineNoChecks(string first, string second, string third)
+        {
+            if (first.Length == 0)
```

Comment:
I won't comment on the rest, but you should use .IsNullOrEmpty on the first check of the arguments